domain/autocert/service/service.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package service
+
+import (
+	"context"
+)
+
+// State describes retrieval and persistence methods for storage.
+type State interface {
+	// Put implements autocert.Cache.Put.
+	Put(ctx context.Context, name string, data []byte) error
+
+	// Get implements autocert.Cache.Get.
+	Get(ctx context.Context, name string) ([]byte, error)
+
+	// Delete implements autocert.Cache.Delete.
+	Delete(ctx context.Context, name string) error
+}
+
+// Logger facilitates emitting log messages.
+type Logger interface {
+	Tracef(string, ...interface{})
+	Warningf(string, ...interface{})
+}
+
+// Service provides the API for working with autocert cache.
+type Service struct {
+	st     State
+	logger Logger
+}
+
+// NewService returns a new service reference wrapping the input state.
+func NewService(st State, logger Logger) *Service {
+	return &Service{
+		st:     st,
+		logger: logger,
+	}
+}
+
+// Put implements autocert.Cache.Put.
+func (s *Service) Put(ctx context.Context, name string, data []byte) error {
+	s.logger.Warningf("storing autocert %s with contents '%+v' in the autocert cache", name, string(data))
```

Comment:
This should be trace I feel and we should drop the data logging.

Reply:
forgot to set it back to trace, done